packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -276,6 +276,29 @@ export const getCurrentChannel: (state: GlobalState) => Channel = createSelector
     },
 );
 
+export const getCurrentChannelNameForSearchShortcut: (state: GlobalState) => string | undefined = createSelector(
+    getAllChannels,
+    getCurrentChannelId,
+    (state: GlobalState): UsersState => state.entities.users,
+    getTeammateNameDisplaySetting,
+    (allChannels: IDMappedObjects<Channel>, currentChannelId: string, users: UsersState, teammateNameDisplay: string): string | undefined => {
+        const channel = allChannels[currentChannelId];
+
+        // Only get the extra info from users if we need it
+        if (channel?.type === Constants.DM_CHANNEL) {
+            const dmChannelWithInfo = completeDirectChannelInfo(users, teammateNameDisplay, channel);
+            return `@${dmChannelWithInfo.display_name}`;
+        }
+
+        // Replace spaces in GM channel names
+        if (channel?.type === Constants.GM_CHANNEL) {
+            return `@${channel.display_name.replace(/ /g, '')}`;
```

Comment:
```suggestion
            return `@${channel.display_name.replace(/\s/g, '')}`;
```

Filter for all possible whitespace characters